README.md

```diff
@@ -48,76 +31,68 @@ You need to have the following installed:
 If you want to benchmark Java, you will also need:
   * A Java SDK 7 (`openjdk-7-jdk` package in Debian)
 
-Note that to use pinning on Linux, `cset shield` must be in a working state.
-Some Linux distributions have been known to ship with this functionality
-broken. See the `cset` tutorial for information on how to test `cset shield`:
-https://rt.wiki.kernel.org/index.php/Cpuset_Management_Utility/tutorial
+You will need either `sudo` (on Linux) or `doas` (OpenBSD) to be setup.
 
-### Kernel arguments
 
-If you are using a Linux system, you will need to set a kernel arguments to
-disable Intel P-states. If your Linux bootloader is Grub, you can follow these
-steps:
+## Step 2 (Linux only): kernel and OS setup
 
-  * Edit /etc/default/grub (e.g. `sudo gedit /etc/default/grub`)
-  * Add `intel_pstate=disable` to `GRUB_CMDLINE_LINUX_DEFAULT`
-  * Run `sudo update-grub`
+The kernel must also be
+configured to be tickless on all CPU cores except the boot core.
+\laurie{shouldn't we tell people how to do this?}
 
-You can disable Krun's P-state check with `--disable-pstate-check`, however
-this is strongly discouraged for real benchmarking.
+### P-states
 
-### The Krun Linux Kernel
+Benchmarking is at its most accurate when Intel p-states are disabled in
+the kernel. If you are using Grub, this can be achieved as follows:
 
-When Krun is run on Linux it requires a custom Linux Kernel which offers low
-latency access to the `IA32_APERF`, `IA32_MPERF` and `IA32_PERF_FIXED_CTR1`
-MSRs (sadly `IA32_APERF` or `IA32_MPERF` cannot be read from user-space via
-`rdpmc` and `rdmsr` is strictly a ring 0 operation). The kernel must also be
-configured to be tickless on all CPU cores except the boot core.
+  * Edit `/etc/default/grub` so that the `GRUB_CMDLINE_LINUX_DEFAULT`
+    variable has `intel_pstate=disable` added to it.
+  * Run `sudo update-grub`
+
+If you are unable to do this, you can disable Krun's P-state check with
+`--disable-pstate-check`, but be aware that this lessens the quality of the
+resulting benchmarking numbers.
+
+### Performnace counters
 
-Instructions and source code can be found here:
-https://github.com/softdevteam/krun-linux-kernel
+We recommend using our custom Linux kernel found at:
 
-#### Benchmarking on a Stock Linux Kernel
+  https://github.com/softdevteam/krun-linux-kernel
 
-You can run Krun on a stock Linux Kernel, but Krun will be unable to
-collect data from:
+which provides low latency access to the following counters:
 
   * IA32_PERF_FIXED_CTR1 (the core cycle counter)
   * IA32_APERF counts
   * IA32_MPERF counts
 
-Since these are highly useful metrics, we strongly advise against using a stock
-Linux kernel for real benchmarking.
+This kernel should be built and installed as any other version of the Linux
```

Comment:
It would be better to point the user to the docs of the krun-linux-kernel.

This is hard in itself, as github now suggests you don't change the default repo branch any more.

I wonder if we can make a wiki for the krun-linux-kernel docs and point to that?

Reply:
Not sure I follow.

Reply:
Let's not detail how to build the kernel here as that is covered in the krun-linux-kernel repo.

The difficulty is pointing users to it. The docs for the kernel are in branches (not master). From this repo, we can either point them to a specific branch in the krun-linux-kernel repo and continually update the link as we track debian kernel versions, or we can point at the krun-linux-kernel repo landing page and update the default branch when a new version is released.

The problem with the former: It's annoying and will be forgotten.

The problem with the latter:
![untitled](https://user-images.githubusercontent.com/604955/31270372-0d502cfc-aa7c-11e7-96a4-cd1834796b64.png)

Github supports wikis. I wonder if we can add some basic info on how to choose the right branch (and look at the docs within) to a wiki page and link to that instead?

Hope that makes it clearer?

Reply:
That perhaps suggests that we should make the other repo do everything in the master branch? Anyway, whatever you choose, you just need to tell me what to write here.

Reply:
I'm not sure we can do that. The master branch is, well, the master branch from upstream.

Let's not let this block this PR. I am in the process of making binary releases of the kernel. We can point them at this, then configuring tickless mode is moot.

Reply:
OK, so please tell me what to write here. I can't guess.

Reply:
For now, kill the sentence:

> This kernel should be built and installed as any other version of the Linux...

Reply:
Just that? Nothing else?

Reply:
I'm not sure what the end result will be until I've finished experimenting with binary releases of the kernel and wikis/branches in the krun repo. Sorry.

I suggest I come back to this later. We've said elsewhere in the docs that you should use the Krun kernel, right?

Reply:
OK (and "yes").